tsl/src/bgw_policy/policy_utils.c

```diff
@@ -100,3 +102,27 @@ subtract_interval_from_now(Interval *lag, Oid time_dim_type)
 			pg_unreachable();
 	}
 }
+
+const Dimension *
+get_open_dimension_for_hypertable(const Hypertable *ht)
+{
+	int32 mat_id = ht->fd.id;
+	const Dimension *open_dim = hyperspace_get_open_dimension(ht->space, 0);
+	Oid partitioning_type = ts_dimension_get_partition_type(open_dim);
+	if (IS_INTEGER_TYPE(partitioning_type))
+	{
+		/* if this a materialization hypertable related to cont agg
+		 * then need to get the right dimension which has
+		 * integer_now function
+		 */
+
+		open_dim = ts_continuous_agg_find_integer_now_func_by_materialization_id(mat_id);
+		if (open_dim == NULL)
+		{
+			elog(ERROR,
```

Comment:
The code path needs a corresponding test case, if it's possible to add one. Otherwise, please add a comment that it should never happen in practice and is needed only for debugging purposes.

Reply:
To be fair this code was always there, just moved around.